src/app/designs/components/design-modules/design-module/design-module.component.ts

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2021 Carnegie Mellon University. All Rights Reserved. 
+ Released under a MIT (SEI)-style license. See LICENSE.md in the project root for license information.
+*/
+
+import {
+  Component,
+  OnInit,
+  ChangeDetectionStrategy,
+  Input,
+  OnChanges,
+} from '@angular/core';
+import { BehaviorSubject, forkJoin, Observable } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { DesignModuleQuery } from 'src/app/designs/state/design-modules/design-module.query';
+import { DesignModuleService } from 'src/app/designs/state/design-modules/design-module.service';
+import { VariablesQuery } from 'src/app/designs/state/variables/variables.query';
+import { ModuleVariablesResult } from 'src/app/editor/component/module-variables/module-variables.models';
+import {
+  DesignModule,
+  EditDesignModuleCommand,
+  Module,
+  ModuleOutput,
+  Variable,
+} from 'src/app/generated/caster-api/model/models';
+import { ModuleService } from 'src/app/modules/state';
+import { ConfirmDialogService } from 'src/app/sei-cwd-common/confirm-dialog/service/confirm-dialog.service';
+
+@Component({
+  selector: 'cas-design-module',
+  templateUrl: './design-module.component.html',
+  styleUrls: ['./design-module.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DesignModuleComponent implements OnInit, OnChanges {
+  @Input()
+  public designModule: DesignModule;
+
+  @Input()
+  public module: Module;
+
+  public isEditing = false;
+  public showOutputs = false;
+
+  private loadingSubject = new BehaviorSubject(false);
+  public loading$ = this.loadingSubject.asObservable();
+
+  private set loading(val: boolean) {
+    this.loadingSubject.next(val);
+  }
+
+  public variables$: Observable<Variable[]>;
+
+  private outputsSubject = new BehaviorSubject<ModuleOutput[]>([]);
+  public outputs$ = this.outputsSubject.asObservable();
+
+  public designOutputs$;
+
+  private savingSubject = new BehaviorSubject(false);
+  public saving$ = this.savingSubject.asObservable();
+
+  private set saving(val: boolean) {
+    this.savingSubject.next(val);
+  }
```

Comment:
Why not use Akita's loading functionality?

Reply:
This is only if this particular designModule is being saved, so a good candidate for local state imo. Akita's loading is for the entire service. I wouldn't want all the individual designModule cards to show a saving state whenever any one of them were being saved.